src/Orleans.Runtime.Abstractions/Streams/SiloPersistentStreamConfigurator.cs

```diff
@@ -8,9 +8,35 @@
 using System;
 using System.Collections.Generic;
 using System.Text;
+using Orleans.Storage;
 
 namespace Orleans.Streams
 {
+    public class PersistentStreamStorageConfigurationValidator : IConfigurationValidator
+    {
+        private IServiceProvider services;
+        private const string pubsubStoreName = "PubSubStore";
+        private string streamProviderName;
+        public PersistentStreamStorageConfigurationValidator(IServiceProvider services, string streamProviderName)
+        {
+            this.services = services;
+            this.streamProviderName = streamProviderName;
+        }
+
+        public void ValidateConfiguration()
+        {
+            var pubsubOptions = services.GetOptionsByName<StreamPubSubOptions>(this.streamProviderName);
+            if (pubsubOptions.PubSubType != StreamPubSubType.ImplicitOnly)
```

Comment:
It would be better to explicitly check the pubsub systems this check applies to, rather than those it does not.  That would be more extensible if/when more implementations are added.

Reply:
sounds good